File: app-service-comments/src/main/java/com/github/arhor/aws/graphql/federation/comments/service/UserService.java
package com.github.arhor.aws.graphql.federation.comments.service;

import com.github.arhor.aws.graphql.federation.comments.generated.graphql.types.User;

import java.util.UUID;

public interface UserService {

    User findInternalUserRepresentation(UUID userId);

    void createInternalUserRepresentation(UUID userId, UUID idempotencyKey);

    void deleteInternalUserRepresentation(UUID userId, UUID idempotencyKey);
}

File: app-service-posts/src/main/kotlin/com/github/arhor/aws/graphql/federation/posts/service/UserService.kt
package com.github.arhor.aws.graphql.federation.posts.service

import com.github.arhor.aws.graphql.federation.posts.generated.graphql.types.User
import java.util.UUID

interface UserService {
    fun findInternalUserRepresentation(userId: UUID): User
    fun createInternalUserRepresentation(userId: UUID, idempotencyKey: UUID)
    fun deleteInternalUserRepresentation(userId: UUID, idempotencyKey: UUID)
}
